pallets/settlement/src/lib.rs

```diff
@@ -1429,8 +1429,8 @@ impl<T: Config> Module<T> {
                         Error::<T>::MaxNumberOfNFTsPerLegExceeded
                     );
                     Self::ensure_venue_filtering(&mut tickers, nfts.ticker().clone(), &venue_id)?;
-                    let unique_nfts: BTreeSet<&NFTId> = nfts.ids().iter().collect();
-                    ensure!(unique_nfts.len() == nfts.len(), Error::<T>::DuplicatedNFTId);
+                    <Nft<T>>::ensure_no_duplicate_nfts(&nfts)
+                        .map_err(|_| Error::<T>::DuplicatedNFTId)?;
```

Comment:
Since this PR should be included as part of the 5.3 release, we can drop the `DuplicatedNFTId` from the settlement pallet and just return the new duplicated nft error from the nft pallet.

5.3 is still in RC (release candidate mode) so we can make small changes like this.